app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/AuthenticatedAction.scala

```diff
@@ -71,48 +76,97 @@ class AuthenticatedAction @Inject() (
       (selfBaseUrl + successRelativeUrl) -> (selfBaseUrl + failureRelativeUrl)
   }
 
-  @SuppressWarnings(Array("org.wartremover.warts.Product", "org.wartremover.warts.Serializable"))
-  override protected def refine[A](request: MessagesRequest[A]): Future[Either[Result, AuthenticatedRequest[A]]] = {
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
-
-    authorisedFunctions.authorised(ConfidenceLevel.L200).retrieve(Retrievals.nino) {
-      case None =>
-        logger.warn("Could not find NINO")
-        Left(errorHandler.errorResult()(request))
-
-      case Some(nino) =>
-        Right(AuthenticatedRequest(NINO(nino), request))
-    }.recoverWith {
-      case _: NoActiveSession =>
-        Left(Redirect(signInUrl, Map("continue" -> Seq(selfBaseUrl + request.uri), "origin" -> Seq(origin))))
+  private def itmpNameToName(itmpName: ItmpName): Option[Name] =
+    itmpName match {
+      case ItmpName(Some(givenName), _, Some(familyName)) => Some(Name(givenName, familyName))
+      case _                                              => None
+    }
 
-      case _: InsufficientConfidenceLevel =>
-        handleInsufficientConfidenceLevel(request, errorHandler.errorResult()(request)).map(Left(_))
+  private def nonItpmNameToName(retrievalName: RetrievalName): Option[Name] =
+    retrievalName match {
+      case RetrievalName(Some(name), _) => {
+        val Array(forename, surname) = name.split(' ')
```

Comment:
we also need to handle the case where there might be middle names inside this name. We need to look for the first and last element after the split.

Reply:
refactored to handle arbitrary names - went for more verbose logic to make the logic apparent